frontend/src/db/shop_db.ts

```diff
@@ -69,41 +90,31 @@ export class AppInfoDB {
     })
   }
 
-  async get_all(): Promise<AppInfoWithState[]> {
+  async remove_multiple_app_infos(ids: number[]): Promise<void> {
     const db = await this.open()
     return new Promise((resolve, reject) => {
-      const transaction = db.transaction('appInfo', 'readonly')
+      const transaction = db.transaction('appInfo', 'readwrite')
       transaction.onerror = () => reject(transaction.error)
       const store = transaction.objectStore('appInfo')
-      const request = store.openCursor()
-      const result: AppInfoWithState[] = []
-      request.onsuccess = () => {
-        const cursor = request.result
-        if (cursor) {
-          result.push(cursor.value)
-          cursor.continue()
-        }
-        else {
-          resolve(result)
-        }
-      }
+      ids.forEach((id: number) => store.delete(id))
+      transaction.oncomplete = () => resolve()
     })
   }
 
   // Add base64 encoded webxdc to the db.
-  async add_webxdc(webxdc: XDCFile, id: number): Promise<void> {
+  async add_webxdc(webxdc: XDCFile, id: string): Promise<void> {
```

Comment:
For now, you only want to download the most recent version so this is the only one we need to cache for now.
I could add a version number, but I would then have to explicitly remove old caches upon new .xdc file and when using `sendToChat` I would always have to set the version as well which is always the newest one. 
The current solution where only one cache for the app_id is used seems far simpler while achieving the same imo. 
Also, you shouldn't forget that indexeddb is only a kv-store, so adding a version to this function would mean we'd have to concatenate app_id + version or so which feels awkward. (might not be a problem though idk)

Reply:
yes, we only need to cache the most recent version -- but for an `app_id` that is in the cache it's internally now unclear what version it is and that can easily lead to bugs in the face of lost messages or other issues.  If the app index has info about "app_id" pointing to the latest version, you don't know for sure if the cache has that version or some prior one.

Reply:
For any cache, a precise key avoids problems.  If the same key can yield different results depending on timing, it's not a good cache-key and can lead to consistency problems.  You argue that those potential inconsistencies don't matter or don't cause problems because they are embedded in a particular flow of overall state changes.  But this means you need to understand a lot of context to rule out cache inconsistencies.  I've been bitten by caching problems for 2-3 decades so i guess i am a bit less optimistic than you ;)